### ISSUE TITLE
Questions on using Apigee Kickstart Dev Portal

### ISSUE THREAD

**Issue by epowelljr** [apigee/apigee-devportal-kickstart-drupal#148]
I’m still having issues with the Apigee kickstart Drupal portal installation.

I installed the portal on Prem. It was a LAMP installation using MariaDB as the Database. The Sever is Redhat.

1. The portal work fine when a user is just surfacing through the portal page and content. 
2. If you login to the portal you will notice the interface structure breaks. The portal display lists of folder. 
3. When you logout of the portal you will get a 404 error.

A. So, I renamed the .htaccess files located in Drupal root and I noticed I could not login or access any pages. I would get the 404 error. So, I change the name back to .htaccess. Now, portal return back to normal.  I can login to the Portal as well but when I login the I noticed the content and Drupal interface structure breaks. There are lists of folder displaying throughout the portal. If I flush the caches it makes things even worse. The Portal breaks completely and I have to import the database to get the portal to function properly again.

I can not fine a solution to fix the issue when logging into the portal.
Can you help?
Thanks
Edgar

**Comment by jacine** [comment]
Hi @epowelljr,

We're confident this is a server configuration issue, as opposed to an issue with this project.  That's to say that if you installed Drupal alone, without the Kickstart, in your current setup, you would likely face the same issue.  We've been trying to help, across numerous issues/comments across this queue, but it's not clear if you've tried our suggestions.  I've added all of those suggestions here for easy reference.

In addition to what we've replied with previously, I'd just like to add that if I were in your position, I would try the following:

1.  Go through every heading on this page, and check it against your setup: https://www.drupal.org/node/228462.  

2.  If you are having issues with the steps (Apache configuration) linked above, try asking your hosting provider for guidance. They may have documentation/guides for exactly what you're trying to achieve.  They're likely aware of Drupal requirements as well, and may be able to better guide you through the way customers need to make configuration changes on their servers.

3. If you still can't get it working, try starting over.  Take another look through the [System requirements for Drupal 8](https://www.drupal.org/docs/8/system-requirements), and try starting over fresh. 

P.S. It would be really helpful if you could keep the dialog in a single issue.  We really do want to help you.  Please use this issue going forward for anything related to your .htaccess/404 errors.  Thank you!

Good luck @epowelljr!  You can do it.  😄 

---

1.  https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/106#issuecomment-497917193, where we suggested:

> It sounds like an issue with your web server configuration with Drupal clean urls:
> * [Fixing clean URL problems](https://www.drupal.org/docs/8/troubleshooting-drupal-8/fix-drupal-8-clean-urls-problems)
> * [This thread has the same issue of links being 404s](https://groups.drupal.org/node/105954)
> 
> _Originally posted by @cnovak in https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/106#issuecomment-498842102_

2.  https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/136, where we suggested: 

> Hi @epowelljr,
> 
> @cnovak added some information that should be helpful in #106.
> 
> I also found this, which suggests that you may need to configure this outside of `.htaccess` itself: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/5/html/deployment_guide/s1-apache-config 
> 
> Hope it helps!
> Jacine
> 
> _Originally posted by @jacine in https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/136#issuecomment-498844631_

3. https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/117#issuecomment-500906282, where we replied:

> Hi @epowelljr,
> 
> Did you try the suggestions we posted for you in #136?
> 
> Are you sure that you have `mod_rewrite` enabled?  Here is another documentation page that might be useful for you: https://www.drupal.org/node/228462.
> 
> Thanks,
> Jacine
> 
> _Originally posted by @jacine in https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/117#issuecomment-500953483_

> @epowelljr here is a doc on how to get [Drupal 8 running on RHEL](https://www.web3us.com/drupal-how-guides/install-drupal-8-composer-and-drush-9) without our Kickstart.  You can replace step number 7 in this doc with the Kickstart composer project command.
> 
> _Originally posted by @cnovak in https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/117#issuecomment-500964763_

**Comment by cnovak** [comment]
We have commented back on each ticket about steps to try, but have not heard back on any of our suggestions. Like @jacine pointed out this is a configuration issue of your web server but we would like to offer help as much as we can.  Let us know in this ticket of anything you find following our suggestions, we will keep this ticket open for you to comment on.

**Comment by cnovak** [comment]
@epowelljr I thought of another solution that may help you - Pantheon.io will bring up an Apigee Kickstart site for you on their platform.  Go to the [Apigee Pantheon Partnership page](https://pantheon.io/product/apigee-pantheon-partnership), scroll to the bottom and click on "Free Consultation" link, use that to request a Kickstart site to be spun up.  Hope this helps!

**Comment by kscheirer** [comment]
I've followed up with Edgar over email to help resolve his issues, this does not appear to be a problem in Kickstart. I think you can close this issue.

**Comment by epowelljr** [comment]
Do you have a default .htaccess file that I can use for the Apigee kickstart Dev Portal?

We had to make edits to rewrite conditions to git the content to format and display.

-Edgar

**Comment by jacine** [comment]
Hi @epowelljr,

Sure, it’s right here: https://git.drupalcode.org/project/drupal/blob/8.7.3/.htaccess

Thanks,
Jacine

**Comment by epowelljr** [comment]
Is this .htaccess identical to the one that is in the kickstart download for 8.7.3?

**Comment by jacine** [comment]
Yes, it’s the same. The Kickstart downloads the latest Drupal core version directly when you run the `composer create-project apigee/devportal-kickstart-project:8.x-dev MY_PROJECT --no-interaction` command.

**Comment by epowelljr** [comment]
We had to include RewriteCond to get the page content to format and display. 
Do you know what is going? Is the the correct place to make these edits?

**Comment by jacine** [comment]
I rarely ever have to make edits to this file. Did you try going through this list yet? https://www.drupal.org/node/228462

**Comment by epowelljr** [comment]
Yes, but we still had issues.

**Comment by jacine** [comment]
What kind of issues? Are you sure mod_rewrite is enabled and properly configured at the server level? Are you installing in the web root directory or a sub directory? The more information you can provide, the better.

**Comment by epowelljr** [comment]
We have PHP 7.3.5 loaded. I saw come document saying that Apigee and the modules are not supported with PHP 7.3.5 but is support with PHP 7.2

Do know what is correct?

**Comment by jacine** [comment]
> We have PHP 7.3.5 loaded. I saw come document saying that Apigee and the modules are not supported with PHP 7.3.5 but is support with PHP 7.2

I'm not aware of any issues with PHP 7.3.  I've got PHP 7.3.1 running locally, with all the latest Apigee modules, and everything is working fine. PHP 7.3 has been recommended, alongside 7.2, for Drupal since version 8.6.4 (see screenshot below). If you recall where you found this information in the docs, please let us know, so we can make sure to verify/update it. 

<a href="https://www.drupal.org/docs/8/system-requirements/php-requirements" target="_blank"><img width="616" alt="Screen Shot 2019-06-16 at 12 19 59 AM" src="https://user-images.githubusercontent.com/60979/59559017-07984c80-8fcd-11e9-8892-f9d3bfa52678.png"></a>

Maybe you need to adjust your `php.ini` settings? Check out the details on this page: https://www.drupal.org/docs/8/system-requirements/php-requirements. Check that your `memory_limit` is at least `256M` and `max_execution_time` at least `60`.  There is another note on that page about `.htaccess` worth checking too.

Anything else you can share?  Are there any errors in the PHP error log?

**Comment by epowelljr** [comment]
The portal is working without error now. I did increase the memory_limit to 512. I increase the file upload. I was set to 128MB. I noticed some images would not display so I increase it to 300MB, I guess that is a little high, what’s your thoughts?

I did not change the max_execution_time. I will check that now.

Thanks

**Comment by jacine** [comment]
> The portal is working without error now. I did increase the memory_limit to 512. I increase the file upload. I was set to 128MB. I noticed some images would not display so I increase it to 300MB, I guess that is a little high, what’s your thoughts? I did not change the max_execution_time. I will check that now. Thanks

Great! 🎉 I’m so glad to hear it’s all working for you.

Unless you are expecting to have users upload large files, 300MB is kind of high.  The largest source image we have in the repo is about 6MB, but we use responsive images everywhere, so they’re all transformed to smaller sizes when viewed.  That’s probably what you experienced, the initial creation of the smaller image variants created by the server.

**Comment by epowelljr** [comment]
1. Responsive images? How can I convert my images to be of such?
2. Also, do I need to install an additional module to handle registered developers or users? Drupal has built-in capabilities to add roles and users.

**Comment by jacine** [comment]
1. There’s nothing you need to do. This is done for you, upon installation. The Drupal entity display configuration is already using responsive image formatters everywhere. The first time (only once) the site is ever loaded the images generate (like Drupal 7 image cache). This is what I was referring to in my previous comment. Are you seeing images on the site?

2.  No additional modules are needed. Check out the Apigee Edge documentation: https://www.drupal.org/docs/8/modules/apigee-edge

**Comment by epowelljr** [comment]
Yes, all the images are displaying correctly. Apparently, it was something else that was preventing the images from displaying. 
If that is the case, I will lower the file upload.

**Comment by jacine** [comment]
Ok great! Thanks for letting us know @epowelljr. 🙂

**Comment by epowelljr** [comment]
Jacine,

Now that we have resolved the issues with the Dev Portal 404 and .htaccess. 

1. I need to connect the Drupal 8 Dev Portal to the Apigee Management server. 
2. Originally, there was a Drupal 7 Dev Portal connected to the Apigee Management server. Do we need to make any updates to the Management server since we are planning on attaching the Drupal 8 version of the Dev Portal?
3. Also, what about SmarDocs? I have the Apigee API Catalogue installed on the Drupal 8 Dev Portal. 

—Edgar

**Comment by epowelljr** [comment]
Hi Jacine,

I trying to connect the Dev Portal to the Apigee Management Server. I applied the credentials the Dev Portal Apigee Edge Connection Settings form. 

I’m getting the error below:
   
Failed to connect to Apigee Edge. cURL error 7: (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)

Do you have a fix for this?

-Edgar

**Comment by kscheirer** [comment]
Hi Edgar,

You don't need to make any updates to the Apigee Management server. The new Apigee API Catalog module is SmartDocs for Drupal 8. If you already have that installed, then you're all set.

The credentials from your D7 site should work fine for D8 as well. Curl error 7 is `CURLE_COULDNT_CONNECT (7) - Failed to connect() to host or proxy.`, so it sounds like the Apigee Management server is not reachable from your D8 devportal. Is the D8 portal on a different server than the D7 portal maybe? It sounds like you're using the on-premise version of the Apigee Management server, is that correct?

**Comment by epowelljr** [comment]
Got it functioning now. Do you have a link for configuring or using the API Catalog module with the Apigee management server?

Yes I believe we are using the on-premise version.

**Comment by epowelljr** [comment]
Karl,

The Dev Portal and the Apigee Management server are using the same Redhat server at different ports.
This server do not have access to the internet. Do know if that will contribute issues when using the Apigee API Catalog and the Apigee Management server.

-Edgar

**Comment by cnovak** [comment]
You can try making the same call from the server using cURL.  You should get the same response.  
For example, you can see this will return a 401, modify url, username, password, and org to match your Edge configuration:

```
curl -v -u username:password -X GET \
  https://api.enterprise.apigee.com/v1/o/myorg \
  -H 'Accept: application/json' 
```

You can also turn on debugging to get more information.  Go to "Extend" in the admin menu you can turn on the [Apigee Edge Debug module](https://www.drupal.org/docs/8/modules/apigee-edge/monitor-apigee-edge-debug-logs).  That will print out the call being made to the logs.

**Comment by kscheirer** [comment]
Hi Edgar,

Good to hear that its working now. I dont think you'll have any trouble being on the same server. The general docs for using the API Catalog are here: https://www.drupal.org/docs/8/modules/apigee-api-catalog-0/document-your-apis.

Basically
- Create a new API Doc entity
- Upload your OpenAPI spec file
- View the new entity

**Comment by epowelljr** [comment]
Hi Karl,

I noticed another issue that has surfaced.

When login as an administrator, I tried to add an API doc.
I browsed to Content, API Catalog, Add API doc. There is an Image section where you can Add media. I tried to add an image but was not able to add it. I was able to browse and select the image and select to insert the image but it would not add the image for the API doc.

Attached is the screenshot from the error log.
Do you know what is going on and how to fix this?

-Edgar

**Comment by cnovak** [comment]
Hello @epowelljr I do not see any screenshots on this ticket, can you add them by going to the ticket directly and adding them instead of replying by email? 

https://github.com/apigee/apigee-devportal-kickstart-drupal/issues/148

**Comment by epowelljr** [comment]
Attached is the screenshot:
<img width="1436" alt="Screen Shot 2019-06-18 at 9 49 59 PM" src="https://user-images.githubusercontent.com/7342666/59741956-103d8c80-9232-11e9-9603-665f1e631929.png">

**Comment by epowelljr** [comment]
Hi Chris,

Can you check the ticket. I attached the error.

**Comment by jacine** [comment]
Hi @epowelljr,

Thanks for the image.  It's very helpful.  It looks like this is actually a Drupal core bug: https://www.drupal.org/project/drupal/issues/2990664

The fix has been committed, so it should be in the next release.

Thanks,
Jacine

**Comment by epowelljr** [comment]
Hi Utpal,

I need to install the patch listed below.
The module that I need to patch is the media_library. 

I tried to apply the patch but the server do not have the ability to allow patching. Can you load patch tool on the server.

-Edgar

**Comment by epowelljr** [comment]
Hi Jacine,

Can you provide suggestions or a fix for the errors below:

We tried to register a developer in the Dev Portal. When entered all details and tried to create an account it throws below exception. 
 

 

Thanks Jacine!

**Comment by jacine** [comment]
Hello @epowelljr!

Did you forget to attach the details?  You might need to do it directly, instead of via email like before.  

Thanks!
Jacine

**Comment by epowelljr** [comment]
It is now attached.
<img width="1224" alt="Screen Shot 2019-06-20 at 3 11 32 PM" src="https://user-images.githubusercontent.com/7342666/59878547-349e8380-936e-11e9-8ed5-b365677d72d6.png">

Thank you!

**Comment by cnovak** [comment]
@epowelljr those error messages are cut off, you should be able to click on the message and see the details.  However, it seems that this issue is full of a lot of different issues/questions.  Can you open an issue for each problem you have instead of having a long thread about different issues?  I am afraid that persons having these same issues will not be able to follow the problem.  Please open a bug for any issues related to this project and fill out the issue template with full details.

**Comment by epowelljr** [comment]
Hi Cnovak,

This is more info on the bug.

<img width="1274" alt="Screen Shot 2019-06-21 at 9 19 50 AM" src="https://user-images.githubusercontent.com/7342666/59929247-0c626380-9406-11e9-8d2f-28003d669c37.png">

**Comment by cnovak** [comment]
@epowelljr pen a bug ticket and fill out details instead of using this same thread for each issue, thanks
